### PR TITLE
Upgrade Storybook to 8.1

### PR DESCRIPTION
**Pull request by ethriel3695**
Upgrading Storybook so we can use the VTA with this repository.

<details>
  <summary>📦 Published PR as canary version: <code>11.5.0--canary.989.9321600388.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install chromatic@11.5.0--canary.989.9321600388.0
  # or 
  yarn add chromatic@11.5.0--canary.989.9321600388.0
  ```
</details>

### CODE REVIEW

**Review by skitterm — APPROVED** [review]
Things looked good to me, but I'm not well-acquainted with details of Storybook v8, so we'll want at least 1 other reviewer who does. Nice work @ethriel3695!

**Review by andrewortwein — DISMISSED** [review]
This is looking pretty good, but I do see a couple potential problems:

1. I had to manually run `yarn set version stable` to force using yarn 4. I'm not sure what we can do to make that easier on others, but maybe we can add a `packageManager` field to package.json. That way, at least developers using `corepack` will automatically [use the correct version](https://nodejs.org/api/corepack.html#configuring-a-package). We might also want to call it out in our README.
2. After setting to use yarn 4 and installing everything, I was able to build just fine but a few of the tests consistently fail:

```
✔ script to run › test
The CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.

 RUN  v0.34.6 /Users/andrew/Code/chromaui/chromatic-cli
      Coverage enabled with v8

 · bin-src/trace.test.ts (3)
   ✓ announceBuild (3)
   ❯ compareBaseline (7)
     ⠴ finds changed dependency names
     · finds added dependency names
     · finds removed dependency names
     · finds nothing given identical files
     · runs the manifest check on yarn berry lock files successfully
     · does not find yarn berry changed dependency name for set resolution
   ✓ returns 0 when the build is publish only
   ✓ should exit with code 0 when the current branch is skipped
   ✓ should exit with code 6 and stop the build when abortSignal is aborted
   ✓ calls out to npm build script passed and uploads files
   ✓ skips building and uploads directly with storybook-build-dir
   ✓ builds with playwright with --playwright
   ✓ builds with cypress with --cypress
   ✓ passes autoAcceptChanges to the index
   ✓ passes autoAcceptChanges to the index based on branch
   ✓ uses custom DNS server if provided
   ✓ with TurboSnap (1)
   ✓ in CI (5)
   ✓ checks for updates
   ⠙ prompts you to add a script to your package.json
   · does not propmpt you to add a script to your package.json for E2E builds
   · ctx should be JSON serializable
   · should write context to chromatic-diagnostics.json if --diagnostics is passed
   · should not write context to chromatic-diagnostics.json if --diagnostics is not passed
   ✓ returns 0 when the build is publish only
   ✓ should exit with code 0 when the current branch is skipped
   ✓ should exit with code 6 and stop the build when abortSignal is aborted
   ✓ calls out to npm build script passed and uploads files
   ✓ skips building and uploads directly with storybook-build-dir
   ✓ builds with playwright with --playwright
   ✓ builds with cypress with --cypress
   ✓ passes autoAcceptChanges to the index
   ✓ passes autoAcceptChanges to the index based on branch
   ✓ uses custom DNS server if provided
   ✓ with TurboSnap (1)
   ✓ in CI (5)
   ✓ checks for updates
   ✓ prompts you to add a script to your package.json
   ⠙ does not propmpt you to add a script to your package.json for E2E builds
   · ctx should be JSON serializable
   · should write context to chromatic-diagnostics.json if --diagnostics is passed
   · should not write context to chromatic-diagnostics.json if --diagnostics is not passed
   ✓ returns 0 when the build is publish only
   ✓ should exit with code 0 when the current branch is skipped
   ✓ should exit with code 6 and stop the build when abortSignal is aborted
   ✓ calls out to npm build script passed and uploads files
   ✓ skips building and uploads directly with storybook-build-dir
   ✓ builds with playwright with --playwright
   ✓ builds with cypress with --cypress
   ✓ passes autoAcceptChanges to the index
   ✓ passes autoAcceptChanges to the index based on branch
   ✓ uses custom DNS server if provided
   ✓ with TurboSnap (1)
   ✓ in CI (5)
   ✓ checks for updates
   ✓ prompts you to add a script to your package.json
   ⠹ does not propmpt you to add a script to your package.json for E2E builds
   · ctx should be JSON serializable
   · should write context to chromatic-diagnostics.json if --diagnostics is passed
   · should not write context to chromatic-diagnostics.json if --diagnostics is not passed
   ✓ returns 0 when the build is publish only
   ✓ should exit with code 0 when the current branch is skipped
   ✓ should exit with code 6 and stop the build when abortSignal is aborted
   ✓ calls out to npm build script passed and uploads files
   ✓ skips building and uploads directly with storybook-build-dir
   ✓ builds with playwright with --playwright
   ✓ builds with cypress with --cypress
   ✓ passes autoAcceptChanges to the index
   ✓ passes autoAcceptChanges to the index based on branch
   ✓ uses custom DNS server if provided
   ✓ with TurboSnap (1)
   ✓ in CI (5)
   ✓ checks for updates
   ✓ prompts you to add a script to your package.json
   ✓ does not propmpt you to add a script to your package.json for E2E builds
   ✓ ctx should be JSON serializable
   ⠹ should write context to chromatic-diagnostics.json if --diagnostics is passed
   · should not write context to chromatic-diagnostics.json if --diagnostics is not passed
   ✓ returns 0 when the build is publish only
   ✓ should exit with code 0 when the current branch is skipped
   ✓ should exit with code 6 and stop the build when abortSignal is aborted
   ✓ calls out to npm build script passed and uploads files
   ✓ skips building and uploads directly with storybook-build-dir
   ✓ builds with playwright with --playwright
   ✓ builds with cypress with --cypress
   ✓ passes autoAcceptChanges to the index
   ✓ passes autoAcceptChanges to the index based on branch
   ✓ uses custom DNS server if provided
   ✓ with TurboSnap (1)
   ✓ in CI (5)
   ✓ checks for updates
   ✓ prompts you to add a script to your package.json
   ✓ does not propmpt you to add a script to your package.json for E2E builds
   ✓ ctx should be JSON serializable
   ✓ should write context to chromatic-diagnostics.json if --diagnostics is passed
   ⠹ should not write context to chromatic-diagnostics.json if --diagnostics is not passed
   · should upload metadata files if --upload-metadata is passed
stdout | node-src/index.test.ts > should upload metadata files if --upload-metadata is passed
Wrote storybook-out/preview-stats.trimmed.json

 ❯ bin-src/trace.test.ts (3) 2417ms
   ❯ Test trace script from package.json (3) 2417ms
     × returns the default output successfully 983ms
     ✓ outputs directory info when -m expanded is passed 828ms
     × outputs untraced info when --mode expanded is passed with -u and an untraced file 605ms
 ❯ node-src/tasks/initialize.test.ts (7)
   ✓ setEnvironment (1)
   ❯ setRuntimeMetadata (3)
     × sets the build command on the context
     ✓ supports yarn
     × supports pnpm
   ✓ announceBuild (3)
 ✓ node-src/tasks/verify.test.ts (9)
 ❯ node-src/tasks/build.test.ts (18)
   ✓ setSourceDir (4)
   ❯ setBuildCommand (4)
     × sets the build command on the context
     ✓ supports yarn
     × supports pnpm
     ✓ warns if --only-changes is not supported
   ✓ buildStorybook (4)
   ✓ buildStorybook E2E (6)
 ✓ node-src/lib/compareBaseline.test.ts (7) 6833ms
 ✓ node-src/lib/getStorybookInfo.test.ts (12)
 ❯ node-src/lib/getDependencies.test.ts (3) 2857ms
   ❯ getDependencies (3) 2856ms
     × should return a set of dependencies 2372ms
     ↓ should handle checked out manifest and lock files [skipped]
     ✓ should handle historic files 484ms
 ✓ node-src/lib/getOptions.test.ts (19)
 ✓ node-src/tasks/upload.test.ts (18) 331ms
 ✓ bin-src/trim-stats-file.test.ts (1)
 ✓ bin-src/init.test.ts (9)
 ✓ node-src/git/getCommitAndBranch.test.ts (19)
 ✓ node-src/tasks/snapshot.test.ts (5)
 ✓ node-src/lib/getDependentStoryFiles.test.ts (33)
 ✓ node-src/tasks/report.test.ts (2)
 ✓ node-src/lib/logSerializers.test.ts (2)
 ✓ node-src/lib/getConfiguration.test.ts (9)
 ✓ node-src/git/findAncestorBuildWithCommit.test.ts (5)
 ✓ node-src/tasks/gitInfo.test.ts (8)
 ✓ node-src/tasks/prepareWorkspace.test.ts (5)
 ✓ node-src/lib/findChangedDependencies.test.ts (13)
 ✓ node-src/lib/getFileHashes.test.ts (1)
 ✓ node-src/lib/checkStorybookBaseDir.test.ts (4)
 ✓ node-src/lib/utils.test.ts (10)
 ✓ node-src/git/getChangedFilesWithReplacement.test.ts (4)
 ✓ node-src/index.test.ts (31) 810ms
 ✓ isChromatic.test.ts (3)
 ✓ node-src/tasks/auth.test.ts (2)
 ✓ node-src/lib/findChangedPackageFiles.test.ts (35)
 ✓ node-src/git/git.test.ts (9)
 ✓ node-src/lib/getStorybookConfiguration.test.ts (5)
 ✓ node-src/lib/writeChromaticDiagnostics.test.ts (3)
 ↓ node-src/lib/compress.test.ts (2) [skipped]
 ✓ node-src/tasks/storybookInfo.test.ts (1)

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 7 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  bin-src/trace.test.ts > Test trace script from package.json > returns the default output successfully
AssertionError: expected 'BASE Directory: .\nStorybook CONFIG D…' to include '$ ./dist/bin.js trace ./bin-src/ui/me…'
 ❯ bin-src/trace.test.ts:22:20
     20| 
     21|     // Add your assertions based on the expected output or behavior of the script
     22|     expect(output).toContain(scriptCommand);
       |                    ^
     23| 
     24|     // Verify that the output does not contain the expanded output

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/7]⎯

 FAIL  bin-src/trace.test.ts > Test trace script from package.json > outputs untraced info when --mode expanded is passed with -u and an untraced file
AssertionError: expected 'BASE Directory: .\nStorybook CONFIG D…' to include './bin-src/ui/messages/errors/invalidR…'
 ❯ bin-src/trace.test.ts:56:20
     54| 
     55|     expect(output).toContain(untracedFileNote);
     56|     expect(output).toContain(untracedFile);
       |                    ^
     57|   });
     58| });

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/7]⎯

 FAIL  node-src/tasks/build.test.ts > setBuildCommand > sets the build command on the context
AssertionError: expected 'yarn run build:storybook --output-dir…' to deeply equal 'npm run build:storybook -- --output-d…'

- Expected
+ Received

- npm run build:storybook -- --output-dir ./source-dir/ --webpack-stats-json ./source-dir/
+ yarn run build:storybook --output-dir ./source-dir/ --webpack-stats-json ./source-dir/

 ❯ node-src/tasks/build.test.ts:58:30
     56|     await setBuildCommand(ctx);
     57| 
     58|     expect(ctx.buildCommand).toEqual(
       |                              ^
     59|       'npm run build:storybook -- --output-dir ./source-dir/ --webpack-stats-json ./source-dir/'
     60|     );

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/7]⎯

 FAIL  node-src/tasks/build.test.ts > setBuildCommand > supports pnpm
AssertionError: expected 'yarn run build:storybook --output-dir…' to deeply equal 'pnpm run build:storybook --output-dir…'

- Expected
+ Received

- pnpm run build:storybook --output-dir ./source-dir/
+ yarn run build:storybook --output-dir ./source-dir/

 ❯ node-src/tasks/build.test.ts:94:30
     92|     await setBuildCommand(ctx);
     93| 
     94|     expect(ctx.buildCommand).toEqual('pnpm run build:storybook --output-dir ./source-dir/');
       |                              ^
     95|   });
     96| 

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[4/7]⎯

 FAIL  node-src/tasks/initialize.test.ts > setRuntimeMetadata > sets the build command on the context
AssertionError: expected { nodePlatform: 'darwin', …(3) } to deeply equal { …(4) }

- Expected
+ Received

  Object {
    "nodePlatform": StringMatching /darwin|linux|win32/,
    "nodeVersion": "21.2.0",
-   "packageManager": "npm",
+   "packageManager": "yarn",
    "packageManagerVersion": "1.2.3",
  }

 ❯ node-src/tasks/initialize.test.ts:50:33
     48|     await setRuntimeMetadata(ctx);
     49| 
     50|     expect(ctx.runtimeMetadata).toEqual({
       |                                 ^
     51|       nodePlatform: expect.stringMatching(/darwin|linux|win32/),
     52|       nodeVersion: process.versions.node,

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[5/7]⎯

 FAIL  node-src/tasks/initialize.test.ts > setRuntimeMetadata > supports pnpm
AssertionError: expected { nodePlatform: 'darwin', …(3) } to deeply equal { …(4) }

- Expected
+ Received

  Object {
    "nodePlatform": StringMatching /darwin|linux|win32/,
    "nodeVersion": "21.2.0",
-   "packageManager": "pnpm",
+   "packageManager": "yarn",
    "packageManagerVersion": "1.2.3",
  }

 ❯ node-src/tasks/initialize.test.ts:94:33
     92|     await setRuntimeMetadata(ctx);
     93| 
     94|     expect(ctx.runtimeMetadata).toEqual({
       |                                 ^
     95|       nodePlatform: expect.stringMatching(/darwin|linux|win32/),
     96|       nodeVersion: process.versions.node,

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[6/7]⎯

 FAIL  node-src/lib/getDependencies.test.ts > getDependencies > should return a set of dependencies
TypeError: Cannot convert undefined or null to object
 ❯ node-src/lib/getDependencies.test.ts:25:19
     23|     expect(dependencyNames).toEqual(
     24|       expect.arrayContaining([
     25|         ...Object.keys(packageJson.dependencies),
       |                   ^
     26|         ...Object.keys(packageJson.devDependencies),
     27|       ])

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[7/7]⎯

 Test Files  4 failed | 29 passed | 1 skipped (34)
      Tests  7 failed | 307 passed | 3 skipped (317)
   Start at  10:05:43
   Duration  7.49s (transform 774ms, setup 1ms, collect 6.61s, tests 14.05s, environment 4ms, prepare 2.09s)
 ```

**Comment by ethriel3695** [comment]
> This is looking pretty good, but I do see a couple potential problems:
> 
> 1. I had to manually run `yarn set version stable` to force using yarn 4. I'm not sure what we can do to make that easier on others, but maybe we can add a `packageManager` field to package.json. That way, at least developers using `corepack` will automatically [use the correct version](https://nodejs.org/api/corepack.html#configuring-a-package). We might also want to call it out in our README.
> 2. After setting to use yarn 4 and installing everything, I was able to build just fine but a few of the tests consistently fail:
> 

@andrewortwein 
My one concern with setting the `packageManager` as yarn, is that currently the CLI uses `ni` as @ghengeveld pointed out yesterday and specifying a specific package manager overwrites some of our tests that use different package managers like `pnpm` because `yarn` is set in `package.json`

**Comment by andrewortwein** [comment]
> > This is looking pretty good, but I do see a couple potential problems:
> > 
> > 1. I had to manually run `yarn set version stable` to force using yarn 4. I'm not sure what we can do to make that easier on others, but maybe we can add a `packageManager` field to package.json. That way, at least developers using `corepack` will automatically [use the correct version](https://nodejs.org/api/corepack.html#configuring-a-package). We might also want to call it out in our README.
> > 2. After setting to use yarn 4 and installing everything, I was able to build just fine but a few of the tests consistently fail:
> 
> @andrewortwein My one concern with setting the `packageManager` as yarn, is that currently the CLI uses `ni` as @ghengeveld pointed out yesterday and specifying a specific package manager overwrites some of our tests that use different package managers like `pnpm` because `yarn` is set in `package.json`

Oh, is that maybe why some tests were failing for me? I used `yarn set version stable` to ensure that it would build with yarn berry instead of yarn 1.x. Are you doing anything differently? What command are you using to build or run tests from your local CLI?

**Comment by ethriel3695** [comment]
@andrewortwein @ghengeveld 
Updating to yarn berry breaks our ability to let people bring their own package manager for Chromatic in CI.
If you notice the failing tests. Even when I specify a different package manager, it detects yarn and throws an error because of the `packageManager` field that gets added automatically when you run `yarn set version berry` locally.

Any ideas on how we get around this?

**Comment by tmeasday** [comment]
@ethriel3695 maybe those CI tests need to unset the field (ie run `corepack use X`)?

This isn't something that would affect users, the `packageManager` field only has an effect in the top-level `package.json` AFAIK.

**Comment by ethriel3695** [comment]
> @ethriel3695 maybe those CI tests need to unset the field (ie run `corepack use X`)?
> 
> 
> 
> This isn't something that would affect users, the `packageManager` field only has an effect in the top-level `package.json` AFAIK.

@tmeasday the interesting thing is I removed that field and pushed the change and CI still fails.

What I'm really looking to avoid is breaking the chromatic CI process for people who don't use yarn.

I have a yml file that address chromatic specifically with node and when we have the packageManager field, that yml workflow fails.

**Comment by tmeasday** [comment]
> I have a yml file that address chromatic specifically with node and when we have the packageManager field, that yml workflow fails.

Is that in this repo? I think the behaviour of things in this repo is going to be different because the `package.json` is at the root.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -36,9 +36,10 @@ jobs:
 
       - name: Prepare repository
         run: git fetch --unshallow --tags
-
+      - run: corepack enable
+      - run: yarn set version berry
       - name: Install dependencies
-        run: yarn install --ignore-scripts
```
Comment (thafryer) [inline .github/workflows/release.yml:41]: Any particular reason for dropping this? Did you notice a behavior change?

File: .circleci/config.yml [hunk 2]
```diff
@@ -12,22 +12,26 @@ jobs:
...
+      - run: yarn build && git status --porcelain
+
       # unit test
       - run: yarn test
 
       # publish code coverage results
       - run: bash <(curl -Ls https://coverage.codacy.com/get.sh)
 
-      # test if dist is correctly generated from src
-      - run: yarn build && git status --porcelain
```
Comment (thafryer) [inline .circleci/config.yml:33]: Why do we not need this anymore?

File: .github/workflows/smoke-test-windows.yml [hunk 3]
```diff
@@ -14,6 +14,8 @@ jobs:
       - uses: actions/setup-node@v3
         with:
           node-version: lts/*
+      - run: corepack enable
+      - run: yarn set version berry
```
Comment (thafryer) [inline .github/workflows/smoke-test-windows.yml:18]: Are both of these needed?

File: .storybook/preview.tsx [hunk 4]
```diff
@@ -14,7 +16,6 @@ ansiHTML.setColors({
   cyan: '5fb3b2',
 });
 
-chalk.enabled = true;
```
Comment (thafryer) [inline .storybook/preview.tsx:17]: Was this an intention deletion?

File: bin-src/trace.test.ts
Comment (thafryer) [inline bin-src/trace.test.ts:1]: How are these changes related to existing code?

File: node-src/lib/getDependencies.test.ts [hunk 5]
```diff
@@ -22,7 +22,7 @@ describe('getDependencies', () => {
     const dependencyNames = Array.from(dependencies).map((dependency) => dependency.split('@@')[0]);
     expect(dependencyNames).toEqual(
       expect.arrayContaining([
-        ...Object.keys(packageJson.dependencies),
+        ...Object.keys(packageJson.dependencies || {}),
```
Comment (thafryer) [inline node-src/lib/getDependencies.test.ts:25]: This is an antipattern in tests. The expect should be much more concrete about what is expects based on a set of inputs.

**Review by thafryer — CHANGES_REQUESTED** [review]

File: node-src/lib/getDependencies.test.ts (hunk 5 above)
Comment (ethriel3695) [reply node-src/lib/getDependencies.test.ts:25]: @thafryer this is required because we don't have any dependencies in our package.json.
The test fails without an empty object. Yarn Berry removes `dependencies: {}` from the `package.json` if no dependencies exist.

File: bin-src/trace.test.ts
Comment (ethriel3695) [reply bin-src/trace.test.ts:1]: The `ui` directory no longer lives in the `bin-src` directory so these tests are incorrect.

File: node-src/lib/getDependencies.test.ts (hunk 5 above)
Comment (thafryer) [reply node-src/lib/getDependencies.test.ts:25]: So, if an empty object is expected for this specific case, why not just be explicit and use the empty object?

File: .github/workflows/release.yml (hunk 1 above)
Comment (ethriel3695) [reply .github/workflows/release.yml:41]: Jarel and I talked and yarn berry does not support that flag `--ignore-scripts` and we don't think we need it.

File: .circleci/config.yml (hunk 2 above)
Comment (ethriel3695) [reply .circleci/config.yml:33]: This was not removed, but rather, it was placed above the yarn test command for more consistent test runs with less flake.

File: .storybook/preview.tsx (hunk 4 above)
Comment (ethriel3695) [reply .storybook/preview.tsx:17]: I'm adding this back. It was a TypeScript error, but we added a ts-expect-error

File: node-src/lib/getDependencies.test.ts (hunk 5 above)
Comment (ethriel3695) [reply node-src/lib/getDependencies.test.ts:25]: Great callout here. We need to change the target packageJson to something that has dependencies to mimic a real use case.

**Review by thafryer — APPROVED** [review]

**Comment by ghengeveld** [comment]
:rocket: PR was released in [`v11.5.0`](https://github.com/chromaui/chromatic-cli/releases/tag/v11.5.0) :rocket:
